agent_identity/src/document/aggregate.rs

```diff
@@ -102,74 +512,200 @@ pub mod document_tests {
+    async fn test_set_status(
+        document_id: String,
+        did_method: SupportedDidMethod,
+        document: CoreDocument,
+        document_with_verification_method: CoreDocument,
+        document_with_domain_linkage_service: CoreDocument,
+    ) {
+        DocumentTestFramework::with(IdentityServices::default())
+            .given(vec![
+                DocumentEvent::DocumentCreated {
+                    document_id: document_id.clone(),
+                    did_method,
+                    document,
+                    status: Status::SignAndValidate,
+                    with_fixed_algorithm: None,
+                },
+                DocumentEvent::PublicKeyUpdated {
+                    document_id: document_id.clone(),
+                    document: document_with_verification_method.clone(),
+                },
+                DocumentEvent::ServiceAdded {
+                    document_id: document_id.clone(),
+                    document: document_with_domain_linkage_service,
+                },
+            ])
+            .when(DocumentCommand::UpdateDocumentStatus {
+                document_id: document_id.clone(),
+                status: Status::Disabled,
+            })
+            .then_expect_events(vec![DocumentEvent::DocumentStatusUpdated {
+                document_id,
+                status: Status::Disabled,
+            }])
+    }
 }
 
 #[cfg(feature = "test_utils")]
 pub mod test_utils {
-    use agent_secret_manager::secret_manager;
-    use agent_shared::{
-        config::{config, get_preferred_signing_algorithm},
-        from_jsonwebtoken_algorithm_to_jwsalgorithm,
-    };
-    use did_manager::{DidMethod, MethodSpecificParameters};
+    use super::get_properties;
+    use crate::state::DOMAIN_LINKAGE_SERVICE_ID;
+    use agent_shared::config::config;
+    use agent_shared::config::SupportedDidMethod;
     use identity_core::convert::FromJson;
+    use identity_did::CoreDID;
     use identity_document::{
         document::CoreDocument,
         service::{Service, ServiceEndpoint},
     };
+    use identity_iota::verification::jwk::Jwk;
+    use identity_iota::verification::{MethodData, MethodScope, MethodType, VerificationMethod};
     use rstest::*;
     use serde_json::json;
 
     #[fixture]
-    pub fn did_method() -> DidMethod {
-        DidMethod::Web
+    pub fn document_id() -> String {
+        "document_id".to_string()
     }
 
     #[fixture]
-    pub async fn document(did_method: DidMethod) -> CoreDocument {
-        let mut secret_manager = secret_manager().await;
+    pub fn did_method() -> SupportedDidMethod {
+        SupportedDidMethod::Web
+    }
 
-        let method_specific_parameters = matches!(did_method, DidMethod::Web).then(|| MethodSpecificParameters::Web {
-            origin: config().url.origin(),
-        });
+    #[fixture]
+    pub fn document() -> CoreDocument {
+        let controller: CoreDID = "did:web:my-domain.example.org".parse().unwrap();
```

Comment:
Consider changing this to `example.test` please

Reply:
This would require this value to be updated as well in `test-config.yaml`:
```yaml
url: "https://my-domain.example.org"
```

which in turn would require almost all other tests to be updated as well, which I would say is out of scope for this PR.

If you agree I will create a separate issue for it :+1:

Reply:
You're right, I only looked at this very change and didn't fully consider all the consequences it would have. There is a similar PR in **identity-wallet** as well: https://github.com/impierce/identity-wallet/pull/492.